src/SThread.java

```diff
@@ -47,14 +47,16 @@ public void run() {
       // Communication loop
       while ((inputLine = in.readLine()) != null) {
         System.out.println("Client/Server said: " + inputLine);
-        if (inputLine.equals("Bye.")) // exit statement
-        break; //Thomas. termination of connection
         outputLine =
             inputLine; // passes the input from the machine to the output string for the destination
 
         if (outSocket != null) {
           outTo.println(outputLine); // writes to the destination
         }
+
+        if (inputLine.equals("Bye.")) { // exit statement
+          break; //Thomas. termination of connection
+        }
```

Comment:
This was the really important one. Fixes a premature connection termination issue on the part of the router